packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -373,39 +395,63 @@ contract Deploy is Deployer {
+        dii.set(dii.protocolVersionsProxy.selector, mustGetAddress("ProtocolVersionsProxy"));
+        dii.set(dii.opcmProxyOwner.selector, cfg.finalSystemOwner());
 
-        // Fault proofs
-        deployOptimismPortal2();
-        deployDisputeGameFactory();
-        deployDelayedWETH();
-        deployPreimageOracle();
-        deployMips();
-    }
+        if (_isInterop) {
+            di = DeployImplementations(new DeployImplementationsInterop());
+        }
+        di.run(dii, dio);
 
-    /// @notice Deploy all of the implementations
-    function deployImplementationsInterop() public {
-        // TODO: Replace the actions in this function with a call to DeployImplementationsInterop.run()
-        console.log("Deploying implementations");
-        deployL1CrossDomainMessenger();
-        deployOptimismMintableERC20Factory();
-        deploySystemConfigInterop();
-        deployL1StandardBridge();
-        deployL1ERC721Bridge();
+        save("L1CrossDomainMessenger", address(dio.l1CrossDomainMessengerImpl()));
+        save("OptimismMintableERC20Factory", address(dio.optimismMintableERC20FactoryImpl()));
+        save("SystemConfig", address(dio.systemConfigImpl()));
+        save("L1StandardBridge", address(dio.l1StandardBridgeImpl()));
+        save("L1ERC721Bridge", address(dio.l1ERC721BridgeImpl()));
 
         // Fault proofs
-        deployOptimismPortalInterop();
-        deployDisputeGameFactory();
-        deployDelayedWETH();
-        deployPreimageOracle();
-        deployMips();
+        save("OptimismPortal2", address(dio.optimismPortalImpl()));
+        save("DisputeGameFactory", address(dio.disputeGameFactoryImpl()));
+        save("DelayedWETH", address(dio.delayedWETHImpl()));
+        save("PreimageOracle", address(dio.preimageOracleSingleton()));
+        save("Mips", address(dio.mipsSingleton()));
+
+        Types.ContractSet memory contracts = _implsUnstrict();
+        ChainAssertions.checkL1CrossDomainMessenger({ _contracts: contracts, _vm: vm, _isProxy: false });
+        ChainAssertions.checkOptimismPortal2({ _contracts: contracts, _cfg: cfg, _isProxy: false });
+        ChainAssertions.checkOptimismMintableERC20Factory({ _contracts: contracts, _isProxy: false });
+        ChainAssertions.checkDisputeGameFactory({ _contracts: contracts, _expectedOwner: address(0), _isProxy: false });
+        ChainAssertions.checkDelayedWETH({
+            _contracts: contracts,
+            _cfg: cfg,
+            _isProxy: false,
+            _expectedOwner: address(0)
+        });
+        ChainAssertions.checkPreimageOracle({
+            _oracle: IPreimageOracle(address(dio.preimageOracleSingleton())),
+            _cfg: cfg
+        });
+        if (_isInterop) {
+            ChainAssertions.checkSystemConfigInterop({ _contracts: contracts, _cfg: cfg, _isProxy: false });
+        } else {
+            ChainAssertions.checkSystemConfig({ _contracts: contracts, _cfg: cfg, _isProxy: false });
+        }
```

Comment:
For every `deploy_()` function which was deleted below, I add (and if necessary created) a corresponding `ChainAssertions` check here.